clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -173,13 +172,63 @@ public void getUri() throws URISyntaxException, IOException {
         Assert.assertNotNull(u);
     }
 
+    @Test
+    public void testGetFileStatus() throws ApiException, IOException {
+        // existing file
+        Path p = new Path("lakefs://repo/main/exists");
+        ObjectStats os = new ObjectStats();
+        os.setPath("exists");
+        os.checksum(UNUSED_CHECKSUM);
+        os.setPathType(PathTypeEnum.OBJECT);
+        os.setMtime(UNUSED_MTIME);
+        os.setSizeBytes(UNUSED_FILE_SIZE);
+        os.setPhysicalAddress(s3Url("/repo-base/exists"));
+        when(objectsApi.statObject("repo", "main", "exists"))
+                .thenReturn(os);
+        LakeFSFileStatus fileStatus = fs.getFileStatus(p);
+        Assert.assertTrue(fileStatus.isFile());
+        Assert.assertEquals(p, fileStatus.getPath());
+
+        // no file
+        Path noFilePath = new Path("lakefs://repo/main/no.file");
+        ApiException noSuchFileException = new ApiException(HttpStatus.SC_NOT_FOUND, "no such file");
+        when(objectsApi.statObject("repo", "main", "no.file"))
+                .thenThrow(noSuchFileException);
+        when(objectsApi.statObject("repo", "main", "no.file/"))
+                .thenThrow(noSuchFileException);
+        when(objectsApi.listObjects(any(), any(), any(), any(), any(), any()))
+                .thenReturn(new ObjectStatsList().results(Collections.emptyList()).pagination(new Pagination().hasMore(false)));
+        Assert.assertThrows(FileNotFoundException.class, () -> fs.getFileStatus(noFilePath));
+
+        // fake directory
+        Path dirPath = new Path("lakefs://repo/main/dir1/dir2");
+        when(objectsApi.statObject("repo", "main", "dir1/dir2"))
+                .thenThrow(noSuchFileException);
+        ObjectStats dirObjectStats = new ObjectStats();
+        dirObjectStats.setPath("dir1/dir2/");
+        dirObjectStats.checksum(UNUSED_CHECKSUM);
+        dirObjectStats.setPathType(PathTypeEnum.OBJECT);
+        dirObjectStats.setMtime(UNUSED_MTIME);
+        dirObjectStats.setSizeBytes(0L);
+        dirObjectStats.setPhysicalAddress(s3Url("/repo-base/dir12"));
+        when(objectsApi.statObject("repo", "main", "dir1/dir2/"))
+                .thenReturn(dirObjectStats);
+        LakeFSFileStatus dirStatus = fs.getFileStatus(dirPath);
+        Assert.assertTrue(dirStatus.isDirectory());
+        Assert.assertEquals(dirPath, dirStatus.getPath());
+    }
```

Comment:
Need to add another test case - when calling getFileStatus with a non-fake dir we expect an exception

Reply:
testExists_NotExists ?